### PR TITLE
feat: add video flow

### PR DESCRIPTION
**Pull request by ice-hector**
## Description
This PR introduces add video flow

## Additional Notes
Display selected video
Fix the text input position when the keyboard opened
Show visibility options before sending a post/video

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Refactoring
- [ ] Documentation
- [ ] Chore

## Screenshots (if applicable)

https://github.com/user-attachments/assets/616f90c8-64f1-4a12-ba9d-de02c14eb373
Screenshot 2024-11-29 at 10 40 59
Screenshot 2024-11-29 at 10 40 51

### CODE REVIEW

File: lib/app/features/feed/create_post/views/components/post_submit_button/post_submit_button.dart [hunk 1]
```diff
@@ -33,14 +38,49 @@ class PostSubmitButton extends HookConsumerWidget {
...
+            final data = operation.data;
+            if (data is Map<String, dynamic> && data.containsKey(textEditorSingleImageKey)) {
+              imageIds.add(data[textEditorSingleImageKey] as String);
+            } else {
+              operations.add(operation);
+            }
+          }
+
+          await ref.read(createPostNotifierProvider.notifier).create(
+                content: Document.fromDelta(Delta.fromOperations(operations)).toPlainText(),
```
Comment (ice-kreios) [inline lib/app/features/feed/create_post/views/components/post_submit_button/post_submit_button.dart:68]: asking to be sure, does `toPlainText` create a markup content?

File: lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart [hunk 2]
```diff
@@ -40,15 +45,64 @@ class CreatePostModal extends HookConsumerWidget {
...
   final bool showCollapseButton;
 
+  final String? videoPath;
+
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final textEditorController = useQuillController(defaultText: content);
+    final keyboardVisibilityController = KeyboardVisibilityController();
+    final scrollController = useScrollController();
+    final visibilityToolbarKey = GlobalKey();
```
Comment (ice-tychon) [inline lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart:55]: Need to cache this. It's gonna create a new key every build.

File: lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart [hunk 3]
```diff
@@ -40,15 +45,64 @@ class CreatePostModal extends HookConsumerWidget {
...
+    final actionsToolbarKey = GlobalKey();
+    final textInputKey = GlobalKey();
+
+    useEffect(
+      () {
+        final subscription = keyboardVisibilityController.onChange.listen((isVisible) {
+          if (isVisible) {
+            WidgetsBinding.instance.addPostFrameCallback((_) {
```
Comment (ice-tychon) [inline lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart:63]: You don't need to wrap `Future.delayed` with the `addPostFrameCallback`

File: lib/app/features/feed/create_post/views/pages/create_post_modal/components/video_preview_cover.dart [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        filePath,
+        looping: true,
+      ),
+    );
+
+    if (!videoController.value.isInitialized) {
+      return const _VideoPlaceholder();
+    }
+
+    useEffect(
```
Comment (ice-tychon) [inline lib/app/features/feed/create_post/views/pages/create_post_modal/components/video_preview_cover.dart:43]: Looks strange. You create the `videoController` using Riverpod, but dispose it with Hooks. It's better to move disposing to Riverpod provider.

File: lib/app/features/feed/create_post/views/pages/create_post_modal/components/video_preview_duration.dart [hunk 5]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/utils/date.dart';
+
+class VideoPreviewDuration extends StatelessWidget {
+  const VideoPreviewDuration({
+    required this.duration,
```
Comment (ice-kreios) [inline lib/app/features/feed/create_post/views/pages/create_post_modal/components/video_preview_duration.dart:9]: It’s already clear and well done! I just have a small suggestion—what do you think about naming it `durationInSeconds` or perhaps accepting a `Duration` parameter to make it even more intuitive?

**Review by ice-kreios — APPROVED** [review]

File: lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart [hunk 6]
```diff
@@ -13,42 +15,56 @@ import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/
...
   final EventReference? parentEvent;
-
   final EventReference? quotedEvent;
-
   final String? content;
-
   final bool showCollapseButton;
+  final String? videoPath;
+
+  final visibilityToolbarKey = GlobalKey();
```
Comment (ice-tychon) [inline lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart:47]: Use hooks to create global keys. Flutter may create another instance of `CreatePostModal` and keys are gonna be recreated.

**Review by ice-alcides — APPROVED** [review]

**Review by ice-kreios — APPROVED** [review]

**Review by ice-kreios — APPROVED** [review]

**Pull request merged**